src/misc.ts

```diff
@@ -588,29 +592,19 @@ export const findById = <T extends Node>(id: string, list?: T[]) =>
 
 export const COLOR_WARNING = "#FBE5AC";
 export const COLOR_WARNING_DARK = "#3E2F0A";
-type CustomWarningColor = typeof COLOR_WARNING | typeof COLOR_WARNING_DARK;
 
-export const getStatusColor = (
-  status: "error" | "warning" | "info" | "success" | "generic",
-  currentTheme?: DefaultTheme,
-): keyof ThemeTokensValues["colors"]["background"] | CustomWarningColor => {
-  switch (status) {
-    case "error":
-      return "surfaceCriticalDepressed";
-    case "info":
-      return "surfaceBrandDepressed";
-    case "success":
-      return "decorativeSurfaceSubdued2";
-    case "warning":
-      // TODO: use color from new macaw theme when will be ready
-      return currentTheme === "defaultDark"
-        ? COLOR_WARNING_DARK
-        : COLOR_WARNING;
-    case "generic":
-      return "surfaceBrandSubdued";
-    default:
-      return "surfaceBrandSubdued";
-  }
+export const getStatusColor = ({
+  status,
+  currentTheme,
+}: {
+  status: "error" | "warning" | "info" | "success" | "generic";
+  currentTheme: DefaultTheme;
+}) => {
+  const statusHue = getStatusHue(status);
```

Comment:
suggestion: `getStatusHue` is base on color names, I would suggest use colors from theme 
```
case "error":
      return red;
    case "info":
      return blue;
    case "success":
      return green;
    case "warning":
      return yellow;
    case "generic":
      return yellow;
      ```